common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RetryingBlockFetcher.java

```diff
@@ -143,11 +167,22 @@ private void fetchAllOutstanding() {
       logger.error(String.format("Exception while beginning fetch of %s outstanding blocks %s",
         blockIdsToFetch.length, numRetries > 0 ? "(after " + numRetries + " retries)" : ""), e);
 
-      if (shouldRetry(e)) {
```

Comment:
shouldn't we just change this line to `if (shouldRetry(e) && !executorAliveChecker.check())`?

Reply:
> shouldn't we just change this line to `if (shouldRetry(e) && !executorAliveChecker.check())`?
Done.